lib/ansible/modules/cloud/vmware/vmware_drs_group.py

```diff
@@ -0,0 +1,600 @@
+            host_group: True if we want to check hosts, else check vms
+
+        Returns: Bool
+
+        """
+
+        if group_name is None:
+            group_name = self.__group_name
+
+        if cluster_obj is None:
+            cluster_obj = self.__cluster_obj
+
+        list_a = self.__vm_list if host_group is False else self.__host_list
+        list_b = self.__populate_vm_host_list(host_group=host_group)
+
+        # By casting lists as a set, you remove duplicates and order doesn't count. Comparing sets is also much faster and more efficient than comparing lists.
+        if set(list_a) == set(list_b):
+            return False
+        else:
+            return True
+
+    def __create_host_group(self):
+
+        # Check if group exists
+        if self.__group_exists():
+            operation = 'edit'
+            existing_group_obj = self.__get_group_by_name()
+        else:
+            operation = 'add'
+
+        # Check if anything has changed when editing
+        if operation == 'add' or (operation == 'edit' and self.__check_if_vms_hosts_changed(host_group=True)):
+
+            group = vim.cluster.HostGroup()
+            group.name = self.__group_name
+            group.host = self.__host_obj_list
+
+            group_spec = vim.cluster.GroupSpec(info=group, operation=operation)
+            config_spec = vim.cluster.ConfigSpecEx(groupSpec=[group_spec])
+
+            if not self.module.check_mode:
+                task = self.__cluster_obj.ReconfigureEx(config_spec, modify=True)
+                wait_for_task(task)
+
+            self.__set_result(group)
+            self.__changed = True
+
+        if operation == 'edit':
+            self.__set_result(existing_group_obj)
+            self.__msg = "Updated host group %s successfully" % (self.__group_name)
+        else:
+            self.__msg = "Created host group %s successfully" % (self.__group_name)
+
+    def __create_vm_group(self):
+
+        # Check if group exists
+        if self.__group_exists():
+            operation = 'edit'
+            existing_group_obj = self.__get_group_by_name()
+        else:
+            operation = 'add'
+
+        # Check if anything has changed when editing
+        if operation == 'add' or (operation == 'edit' and self.__check_if_vms_hosts_changed()):
+
+            group = vim.cluster.VmGroup()
+
+            group.name = self.__group_name
+            group.vm = self.__vm_obj_list
+
+            group_spec = vim.cluster.GroupSpec(info=group, operation=operation)
+            config_spec = vim.cluster.ConfigSpecEx(groupSpec=[group_spec])
+
+            # Check if dry run
+            if not self.module.check_mode:
+                task = self.__cluster_obj.ReconfigureEx(config_spec, modify=True)
+                wait_for_task(task)
+
+            self.__set_result(group)
+            self.__changed = True
+
+        if operation == 'edit':
+            self.__set_result(existing_group_obj)
```

Comment:
same remark as with the `__create_host_group` on the use of `existing_group_obj`